huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_account_test.go

```diff
@@ -0,0 +1,263 @@
+package rds
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	model "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/rds/v3/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccRdsAccount_basic(t *testing.T) {
+	rName := acceptance.RandomAccResourceName()
+	resourceName := "huaweicloud_rds_account.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckRdsAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testRdsAccount_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRdsAccountExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "password", "Test@12345678"),
+				),
+			},
+			{
+				Config: testRdsAccount_update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRdsAccountExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "password", "Test@123456789"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"password"},
+			},
+		},
+	})
+}
+
+func testAccCheckRdsAccountDestroy(s *terraform.State) error {
+	c := acceptance.TestAccProvider.Meta().(*config.Config)
+	client, err := c.HcRdsV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return fmt.Errorf("Error creating RDS client: %s", err)
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "huaweicloud_rds_account" {
+			continue
+		}
+
+		// Split instance_id and user from resource id
+		parts := strings.SplitN(rs.Primary.ID, "/", 2)
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid id format, must be <instance_id>/<user>")
+		}
+		instance_id := parts[0]
+		user_name := parts[1]
+		// items on every page, [1, 100]
+		limit := int32(100)
+		// List all db users
+		request := &model.ListDbUsersRequest{
+			InstanceId: instance_id,
+			Limit:      limit,
+			Page:       int32(1),
+		}
+
+		for {
+			response, err := client.ListDbUsers(request)
+			if err != nil {
+				return nil
+			}
+			users := *response.Users
+			if len(users) == 0 {
+				break
+			} else {
+				request.Page += 1
+				for _, user := range users {
+					if user.Name == user_name {
+						return fmt.Errorf("Rds account still exists")
+					}
+				}
+			}
```

Comment:
suggest to bring these codes out of `else`.

Reply:
If moving this out of else, we need to save all users and then use a new loop for that to find the target one. This can make it break in time in one loop which will be more effective.